app/jobs/publication_download_job.rb

```diff
@@ -6,22 +6,26 @@ class PublicationDownloadJob < ApplicationJob
   def perform(file_id)
     file = ActivityInsightOAFile.find(file_id)
 
-    Net::HTTP.start(ActivityInsightOAFile::S3_AUTHORIZER_HOST_NAME, 80) do |http|
+    Net::HTTP.start(ActivityInsightOAFile::S3_AUTHORIZER_HOST_NAME, use_ssl: true) do |http|
       request = Net::HTTP::Get.new file.download_uri
       request['X-API-Key'] = Settings.activity_insight_s3_authorizer.api_key
 
       http.request request do |response|
-        unless File.directory?(file.file_download_location.store_dir)
-          FileUtils.mkdir_p(file.file_download_location.store_dir)
-        end
-        File.open(file.file_download_location.store_dir.join(file.download_filename), 'w:ASCII-8BIT') do |io|
-          response.read_body do |chunk|
-            io.write chunk
+        if response.code == '200'
+          unless File.directory?(file.file_download_location.store_dir)
+            FileUtils.mkdir_p(file.file_download_location.store_dir)
+          end
+          File.open(file.file_download_location.store_dir.join(file.download_filename), 'w:ASCII-8BIT') do |io|
+            response.read_body do |chunk|
+              io.write chunk
+            end
           end
+
+          file.update_download_location
+        else
+          Rails.logger.error "#{response.code}: #{response.message}"
```

Comment:
@ajkiessl - Scratch that last comment. Rescuing in the service won't do any good since the job is run async. So I think maybe we continue to log the error in the job where you've doing it now, but we'll also need to set `downloaded: false` on the file record at that point.

Reply:
Ah, nice catch.  Thank you, I'll fix this.

Reply:
👍
I don't think that we're currently doing anything with that `downloaded: false` state, but it makes it easy to query for all of the failed file downloads. In addition to logging the errors, maybe it would be helpful to have a list of these in the UI somewhere too? Maybe not. That's something to consider though since files in this state will essentially be 'stuck' at this point in the workflow, and they'll require some kind of action to move forward - for example, if the failure was due to a temporary Activity Insight outage, then we'd want to clear the downloaded flag after the outage was over so the the download could be retried.

In any case, that could be a separate issue and doesn't need to be a part of this PR.

Reply:
We may also find that, in addition to handling bad HTTP response codes, it might make sense to handle `Net::HTTP` errors (timeout errors and such) here in the same way. That way we don't also need to check for failed jobs in order to find all of the downloads that didn't work. But we should probably wait and see what kinds of problems actually happen first...

Reply:
Ok, moving the conversation over here: #779